cpp/src/parquet/encoding_test.cc

```diff
@@ -882,6 +894,12 @@ TYPED_TEST(EncodingAdHocTyped, PlainArrowDirectPut) {
   }
 }
 
+TYPED_TEST(EncodingAdHocTyped, PlainArrowDirectPutMultiRound) {
+  for (auto seed : {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}) {
```

Comment:
A random test is great but it would be nice to have a direct test which demonstrates the regression and fix.

Reply:
Actually using `round >= 1` point out the error case

Reply:
Randomized tests should be a backup.  If there is a known regression, lets add a test for it.

Reply:
Added as `TestPlainBooleanArrayEncoding.AdHocRoundTrip`